server/src/main/kotlin/com/kamelia/hedera/core/Response.kt

```diff
@@ -53,10 +52,50 @@ data class ResultData<T>(
     val data: T? = null
 )
 
-suspend inline fun ApplicationCall.respond(response: Response<*, *>) {
-    val res: suspend (ResultData<*>) -> Unit = { result ->
-        this.response.status(response.status)
-        result.data?.let { respond(it) }
-    }
-    response.ifSuccessOrElse(res, res)
+suspend inline fun <reified S, reified E> ApplicationCall.respond(response: Response<S, E>) {
+    response.ifSuccessOrElse(
+        { result ->
+            this.response.status(response.status)
+            result.data?.let { respond(it) }
+        },
+        { result ->
+            this.response.status(response.status)
+            result.data?.let { respond(it) }
+        }
+    )
+}
```

Comment:
more than an idea, it's now necessary haha. Because ktor's respond function is now inline and reified, if we pass it a wildcard generic object (which was the case before), it will think it's of type Any and wont be able to serialize it. So now we need to propagate that and give it the actual type, and for that we need to inline reify the generics